frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,327 @@
+          <Button
+            variant="link"
+            onClick={(event) => {
+              event.preventDefault();
+              event.stopPropagation();
+              setFilterText('');
+              filterRef.current?.focus();
+            }}
+            className="co-namespace-selector__clear-filters"
+          >
+            {t('Clear filters')}
+          </Button>
+        </EmptyStatePrimary>
+      </EmptyState>
+    </>
+  );
+
+  const filteredOptions = (favorites?: boolean) =>
+    options.filter((option) => isOptionShown(option, favorites));
+
+  const menuGroup = (favorites?: boolean) => {
+    const label = favorites ? t('public~Favorites') : t('public~Projects');
+
+    if (filteredOptions(favorites).length === 0) {
+      return favorites ? null : noResults;
+    }
+
+    const menuItems = filteredOptions(favorites).map((option) => {
+      return (
+        <MenuItem
+          translate="true"
+          key={option.key}
+          itemId={option.key}
+          isFavorited={isFavorite(option)}
+          isSelected={selected === option.key}
+          isActive={activeOption === option.key}
+        >
+          {option.title}
+        </MenuItem>
+      );
+    });
+    return (
+      <>
+        <Divider />
+        <MenuGroup label={label} translate="true">
+          <MenuContent translate="true">
+            <MenuList translate="true">{menuItems}</MenuList>
+          </MenuContent>
+        </MenuGroup>
+      </>
+    );
+  };
+
+  const actionList = (
+    <MenuList translate="true">
+      {actions.map((action) => {
+        return (
+          <div className="co-namespace-selector__footer-item" key={action.actionKey}>
+            <Button
+              variant="secondary"
+              onClick={(event) => {
+                setMainSelectOpen(false);
+                onSelect(event, action.actionKey);
+              }}
+              className="co-namespace-selector__action-button"
+            >
+              {action.actionTitle}
+            </Button>
+          </div>
+        );
+      })}
+    </MenuList>
+  );
+
+  const footer =
+    actions?.length > 0 ? (
+      <Menu
+        onSelect={(event: React.MouseEvent, itemId: string) => {
+          setMainSelectOpen(false);
+          onSelect(event, itemId);
+        }}
+        onGetMenuHeight={(menuId, height) => setFooterHeight(height)}
+        className="co-namespace-selector__footer"
+      >
+        {actionList}
+      </Menu>
+    ) : null;
+
+  const NamespaceChildSelect = (
+    <div
+      style={{
+        height: MAX_HEIGHT,
+      }}
+      className="co-namespace-selector__menu"
+      ref={menuRef}
+    >
+      <Menu
+        onSelect={(event: React.MouseEvent, itemId: string) => {
+          setMainSelectOpen(false);
+          onSelect(event, itemId);
+        }}
+        onActionClick={onFavorite}
+        style={{
+          maxHeight: `calc(${MAX_HEIGHT} - ${footerHeight}px)`,
+          overflow: 'auto',
+        }}
```

Comment:
If we use MenuFooter, we shouldn't need this inline style calculation.

Reply:
I'll be changing how the menu is constructed to move the switch to be under the filter text.  Will try to avoid inline styles.